nirc_ehr/resources/queries/dbo/q_protocol.sql

```diff
@@ -34,6 +34,44 @@ SELECT 'PROTOCOL-' || PROTOCOL_ID AS "objectid",
        STOCK_YN AS "isStock",
        SUBMISSION_DATE AS "submissionDate",
        REVIEW_COMPLETION_DATE AS "reviewCompletionDate",
-       HONOR_RESET_YN AS "isHonorReset"
-FROM PROTOCOL
+       HONOR_RESET_YN AS "isHonorReset",
+       COALESCE(MAX(CAST(adt.CHANGE_DATETIME AS TIMESTAMP)), to_date('01/01/1970' ,'MM/DD/YYYY')) AS modified
+FROM PROTOCOL p
+LEFT JOIN AUDIT_TRAIL adt ON p.PROTOCOL_ID = substring(PRIMARY_KEY_VALUES, length('PROTOCOL_ID = '))
+AND adt.TABLE_NAME = 'PROTOCOL'
 WHERE PROTOCOL_NUMBER IS NOT NULL
+GROUP BY PROTOCOL_ID,
```

Comment:
Long GROUP BYs are bad for query perf and maintenance.

Here, you are only joining to AUDIT_TRAIL to get the MAX(adt.CHANGE_DATETIME). There's no need to do the left join (expanding the row count based on the AUDIT_TRAIL rows) and then collapse it back (which forces the DB to sort on ALL of the fields in the GROUP BY).

Instead, make this part of the LEFT JOIN. Don't simply join AUDIT_TRAIL, join to a subquery that does a GROUP BY on substring(PRIMARY_KEY_VALUES, length('PROTOCOL_ID = ')) and SELECTs the substring and the MAX(adt.CHANGE_DATETIME)

Reply:
done